zerver/lib/timestamp.py

```diff
@@ -49,9 +53,35 @@ def datetime_to_timestamp(dt: datetime) -> int:
     return int(dt.timestamp())
 
 
+@cache
+def get_date_time_pattern_generator(language: str) -> icu.DateTimePatternGenerator:
+    return icu.DateTimePatternGenerator.createInstance(icu.Locale(language))
+
+
+@cache
+def get_icu_time_zone(time_zone: str) -> icu.TimeZone:
+    return icu.TimeZone.createTimeZone(time_zone)
+
+
+@cache
+def get_date_time_format(language: str, use_twenty_four_hour_time: bool) -> icu.SimpleDateFormat:
+    skeleton = f"yMMMEd{'H' if use_twenty_four_hour_time else 'h'}mz"
```

Comment:
This changes the (en-US) format from “Saturday, February 03, 2001 at 04:05 AM PST” to “Sat, Feb 3, 2001, 4:05 AM PST” to more closely match `timerender.format_markdown_time` (however, the latter doesn’t include “PST”).

We have a little flexibility here, as long as we remain within the skeleton patterns that CLDR supports: https://cldr.unicode.org/translation/date-time/date-time-patterns#additional-date-time-formats